src/trpc/router/restRouter.ts

```diff
@@ -205,6 +238,255 @@ export const restRouter = router({
             return userMethods.edit({ id: user.id, ...input.data });
         }),
 
+    editUserByLogin: restProcedure
+        .meta({
+            openapi: {
+                method: 'PUT',
+                path: '/users/{login}',
+                protect: true,
+                summary: 'Update user by login',
+            },
+        })
+        .input(
+            z.object({
+                login: z.string(),
+                data: z.object({
+                    surname: z.string(),
+                    firstName: z.string(),
+                    middleName: z.string().optional(),
+                    registrationEmail: z.string().optional(),
+                    phone: z.string().optional(),
+                    organizationUnitId: z.string().optional(),
+                    supervisorLogin: z.string().optional(),
+                }),
+            }),
+        )
+        .output(
+            z.object({
+                id: z.string(),
+                name: z.string().nullable(),
+                login: z.string().nullable(),
+                registrationEmail: z.string(),
+                corporateEmail: z.string(),
+                active: z.boolean(),
+                supervisorLogin: z.string().nullish(),
+            }),
+        )
+        .query(async ({ input, ctx }) => {
+            const user = await userMethods.getByLogin(input.login);
+            let newSupervisor: Awaited<ReturnType<typeof userMethods.getByLogin>> | null = null;
+
+            if (user.supervisor?.login !== input.data.supervisorLogin && input.data.supervisorLogin) {
+                newSupervisor = await userMethods.getByLogin(input.data.supervisorLogin);
+            }
+
+            const fieldsToUpdate: Record<string, string> = {};
```

Comment:
Let's create a new ticket for expanding userMethods.create. All this code can be moved to a method

Reply:
https://github.com/taskany-inc/crew/issues/751